sw/device/lib/crypto/drivers/kmac.h

```diff
@@ -259,6 +277,130 @@ OT_WARN_UNUSED_RESULT
+                            crypto_uint8_buf_t *digest);
+
+/**
+ * Compute SHAKE-256 in one-shot.
+ *
+ * Warning: The caller must ensure that `digest->len` contains the
+ * requested digest length (in bytes).
+ *
+ * The caller must ensure that `message` and `digest` have properly
+ * allocated `data` fields whose length matches their `len` fields.
+ *
+ * @param message The input message.
+ * @param digest The digest buffer to return the result.
+ * @return Error status.
+ */
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_shake_256(crypto_const_uint8_buf_t message,
+                            crypto_uint8_buf_t *digest);
+
+/**
+ * Compute CSHAKE-128 in one-shot.
+ *
+ * Warning: The caller must ensure that `digest->len` contains the
+ * requested digest length (in bytes).
+ *
+ * The caller must ensure that `message` and `digest` have properly
+ * allocated `data` fields whose length matches their `len` fields.
+ *
+ * @param message The input message.
+ * @param func_name The function name.
+ * @param cust_str The customization string.
+ * @param digest The digest buffer to return the result.
+ * @return Error status.
+ */
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_cshake_128(crypto_const_uint8_buf_t message,
+                             crypto_const_uint8_buf_t func_name,
+                             crypto_const_uint8_buf_t cust_str,
+                             crypto_uint8_buf_t *digest);
+
+/**
+ * Compute CSHAKE-256 in one-shot.
+ *
+ * Warning: The caller must ensure that `digest->len` contains the
+ * requested digest length (in bytes).
+ *
+ * The caller must ensure that `message` and `digest` have properly
+ * allocated `data` fields whose length matches their `len` fields.
+ *
+ * @param message The input message.
+ * @param func_name The function name.
+ * @param cust_str The customization string.
+ * @param digest The digest buffer to return the result.
+ * @return Error status.
+ */
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_cshake_256(crypto_const_uint8_buf_t message,
+                             crypto_const_uint8_buf_t func_name,
+                             crypto_const_uint8_buf_t cust_str,
+                             crypto_uint8_buf_t *digest);
+
+/**
+ * Compute KMAC-128 in one-shot.
+ *
+ * Warning: The caller must ensure that `digest->len` contains the
+ * requested digest length (in bytes).
+ *
+ * The caller must ensure that `message` and `digest` have properly
+ * allocated `data` fields whose length matches their `len` fields.
+ *
+ * @param key_share0 The first share of the key.
+ * @param key_share1 The second share of the key.
+ * @param key_len The size of each share in bytes.
+ * @param message The input message.
+ * @param func_name The function name.
+ * @param cust_str The customization string.
+ * @param digest The digest buffer to return the result.
+ * @return Error status.
+ */
+OT_WARN_UNUSED_RESULT
+kmac_error_t kmac_kmac_128(uint32_t *key_share0, uint32_t *key_share1,
```

Comment:
shouldn't the keys be `const`?

Reply:
Agreed, these should be `const`. Since I am defining a new struct for passing key, it eventually looks like this:
```
OT_WARN_UNUSED_RESULT
kmac_error_t kmac_kmac_128(kmac_blinded_key_t *key,
                           crypto_const_uint8_buf_t message,
                           crypto_const_uint8_buf_t cust_str,
                           crypto_uint8_buf_t *digest);
```
where the struct has `const` declaration:
```
typedef struct kmac_blinded_key {
  const uint32_t *share0;
  const uint32_t *share1;
  // The length of single share in bytes
  size_t len;
} kmac_blinded_key_t;
```